Feature/common function

Jeshan7 opened the pull request:
Changes made in this PR

- Added a common Function **getCoutryInfo** to access all the functions present inside the library.
- Also wrote test cases for the common function.
- Divided the project into three module 
   - functions.js - containing all the utility functions
   - index.js - contains the common function
   - map.js - contains a map to access required function

Jeshan7 committed:
Changes made in this commit
- Added a common function
- Implemented factory pattern
- Added a map to store functions reference
- Moved all the functions in separate module

Jeshan7 committed:
- Added test cases for the common funtion

Jeshan7 committed:
Changes made in this commit
    - Added functionality of fetching conditional data for specific country
    as well as for all countries
    - Added a test for fetching conditional data for specific country

akshay1malta commented on src/index.js:
```diff
@@ -1,77 +1,33 @@
-};
-
-exports.getCurrencybyCountryCode = (countryCode, options) => {
-   return currencyData[countryCode.toUpperCase()];
```
can you please add those

akshay1malta commented on src/index.js:
```diff
@@ -1,77 +1,33 @@
+      });
+      return newArr;
+    }
+  } else {
```
there is no need of else as above if is returning Array, it wont come to the else condition

akshay1malta commented on src/index.js:
```diff
@@ -1,77 +1,33 @@
+  if (Array.isArray(methodType)) {
+    if (attributes) {
+      let countryObj = countriesData.filter((country) => {
+        return country.country_name.toLowerCase() == attributes.toLowerCase();
```
attributes.toLowerCase(); it will break if attributes is number

akshay1malta commented on src/index.js:
```diff
@@ -1,77 +1,33 @@
+        let newObj = {};
+        for (let item of methodType) {
+          new_obj[item.toLowerCase()] = data[item.toLowerCase()];
+        }
```
can you please make a common function as it is repetative with above one

Jeshan7 committed:
Changes made in this commit
  - Added new test case
  - Added common function for a case in getCountryinfo function

Jeshan7 committed:
Merge branch 'main' into feature/common-function
